src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs

```diff
@@ -0,0 +1,638 @@
+                }
+            }
+            channel.Writer.Complete();
+
+            await callbackTask;
+        }
+
+        /// <summary>
+        /// Validates each aspect of a <see cref="LogEntry"/> compared to the expected values
+        /// on a reference <see cref="LogEntry"/> instance.
+        /// </summary>
+        /// <remarks>
+        /// The Exception property of the <paramref name="expected"/> argument is compared
+        /// to the first line of the Exception property of the <paramref name="actual"/> argument,
+        /// which contains the exception name and message.
+        /// </remarks>
+        private static void ValidateEntry(LogEntry expected, LogEntry actual)
+        {
+            Assert.Equal(expected.Category, actual.Category);
+
+            Assert.Equal(expected.EventId, actual.EventId);
+            
+            Assert.Equal(expected.EventName, actual.EventName);
+
+            if (null == expected.Exception)
+            {
+                Assert.Null(actual.Exception);
+            }
+            else
+            {
+                Assert.NotNull(actual.Exception);
+                // Only compare the first line, which contains the exception type and message.
+                // The other lines contain callstack information, which will vary between machines.
+                string firstLine = actual.Exception.Split(Environment.NewLine)[0];
+                Assert.Equal(expected.Exception, firstLine);
+            }
+
+            Assert.Equal(expected.LogLevel, actual.LogLevel);
+
+            Assert.Equal(expected.Message, actual.Message);
+
+            // TODO: Work on scope verification
+            if (null == expected.Scopes)
+            {
+                Assert.Null(actual.Scopes);
+            }
+            else
+            {
+                Assert.NotNull(actual.Scopes);
+                Assert.Equal(expected.Scopes.Count, actual.Scopes.Count);
+                foreach (string expectedKey in expected.Scopes.Keys)
+                {
+                    Assert.True(actual.Scopes.TryGetValue(expectedKey, out JsonElement? actualValue), $"Expected Scopes to contain '{expectedKey}' key.");
+                    Assert.Equal(expected.Scopes[expectedKey], actualValue);
+                }
+            }
+
+            Assert.NotNull(actual.State);
+            Assert.Equal(expected.State.Count, actual.State.Count);
+            foreach (string expectedKey in expected.State.Keys)
+            {
+                Assert.True(actual.State.TryGetValue(expectedKey, out string actualValue), $"Expected State to contain '{expectedKey}' key.");
+                Assert.Equal(expected.State[expectedKey], actualValue);
+            }
+        }
+
+        private static LogEntry CreateTraceEntry(string category)
+        {
+            return new LogEntry()
+            {
+                Category = category,
+                EventId = 1,
+                EventName = string.Empty,
```

Comment:
All of the log events being received by the tests show an empty event name for some reason. I will investigate this and try to correct it in this PR.

Reply:
Many events do not have names, but I am surprised none of them do.

Reply:
All of the log events that are being generated in the logger scenario are explicitly written by the test app and have Event IDs with both ID and name values.

Reply:
Fix for this is https://github.com/dotnet/diagnostics/pull/2291